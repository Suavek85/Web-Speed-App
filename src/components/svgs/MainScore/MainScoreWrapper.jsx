import React from "react"
import { useSelector } from "react-redux"
import MainScoreUI from './MainScoreUI'
import calcPerformanceAverage from '../../../helpers/calcPerformanceAverage'
import { appStates } from '../../../constants/states'

function WebOptimizeWrapper() {
  const getAppState = useSelector(state => state.stateReducer.getAppState)
  const loadingStatus = getAppState === appStates.LOADING

  const getLighthouseData = useSelector(state => state.stateReducer.getLighthouseData)
  const getScore = getLighthouseData.map(el => el.score)
  const averageScore = calcPerformanceAverage(getScore)
  
  return <MainScoreUI 
    getAppState={getAppState} 
    loadingStatus={loadingStatus}
    averageScore={averageScore} 
  />
}

export default WebOptimizeWrapper

