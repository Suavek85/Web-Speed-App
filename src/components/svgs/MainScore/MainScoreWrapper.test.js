import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import MainScoreWrapper from './MainScoreWrapper'

it('should match the snapshot', () => {
  const spyUseSelector = jest.spyOn(redux, 'useSelector')
  spyUseSelector.mockReturnValue('INACTIVE')
  const spyUseSelectorLightHouse = jest.spyOn(redux, 'useSelector')
  spyUseSelectorLightHouse.mockReturnValue(null)
  const wrapper = shallow(<MainScoreWrapper />)
  expect(wrapper).toMatchSnapshot()
})