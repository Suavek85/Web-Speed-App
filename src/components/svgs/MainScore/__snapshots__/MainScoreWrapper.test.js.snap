// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<MainScoreUI
  averageScore={NaN}
  getAppState={null}
  loadingStatus={false}
/>
`;
